src/emojis/emojiHandler.ts

```diff
@@ -0,0 +1,36 @@
+import emojis from "../helpers/files/emojis.json";
+
+export function getEmojisToReactWith(text: string): Array<string> {
+    const lowerCaseText = text.toLowerCase();
+
+    const keywords = Object.values(emojis).flat();
+
+    // search for each keyword in the text
+    const emojisToReactWith = findMatchingEmojiKeywords(
+        keywords,
+        lowerCaseText,
+        emojis
+    );
+
+    return Array.from(new Set(emojisToReactWith));
+}
+
+function findMatchingEmojiKeywords(
+    keywords: string[],
+    lowerCaseText: string,
+    emojis: Record<string, string[]>
+): Array<string> {
+    const matchingEmojiKeywords: Array<string> = [];
+
+    for (const keyword of keywords) {
```

Comment:
I was confused on this one as we are already getting all the emoji keywords at line 6, why do we need to iterate over all these again in `findMatchingEmojiKeywords`? Because, it does not handle multiple matches, that's why. 😄😄

Reply:
Yeah it's bit confusing. Thus, I have splitted the large file into smaller chunks. This will make it more clear and easy to understand/refactor.

But, for your question, line 6 returns all the keywords from emojis.json

Reply:
Feel free to refactor/update once it's merged Ashish.

One thing we know at the moment, for the given keyword, it doesn't return multiple matches as you mentioned!

Reply:
Ohh sure sure dai, I don't know the context properly, so i was little confused. Thanks for clarifying!😀